lib/plug/adapters/test/conn.ex

```diff
@@ -87,6 +87,11 @@ defmodule Plug.Adapters.Test.Conn do
     {tag, data, %{state | req_body: rest}}
   end
 
+  def push(%{owner: owner, ref: ref}, path, headers) do
+    send owner, {ref, {:push, path, headers}}
```

Comment:
We can either do that or we can store it in private and have a helper in Plug.ConnTest to read it out of private. I would prefer to store it in private because it forces people to pass the connectino around. Thoughts?

Reply:
@josevalim When you say store it in private, you mean on the state for the test adapter? Not private as in conn.private?

Reply:
I mean conn.private but you are right, it needs to be the state of the test adapter.